src/main/java/views/add_edit_epense_views/EditExpenseV.java

```diff
@@ -102,27 +104,36 @@ public void actionPerformed(ActionEvent evt) {
                 if (recurExpense.getName().equals(selectedExpense)) {
                     // Set isRecurringCheckBox checked if the expense being edited was a recurring expense
                     isRecurringCheckBox.setSelected(true);}}
-        }
+        } else if (selectedExpense == null) {
+            JOptionPane.showMessageDialog(this, "Please select an expense you wish to edit.");}
 
-        if(selectedExpense == null){
-            JOptionPane.showMessageDialog( this, "Please select an expense you wish to edit.");
-        }else if(evt.getSource() == submit){
-            String selectedCategory = (String) categoryCombo.getSelectedItem();
-            boolean isRecurring = isRecurringCheckBox.isSelected();
-            ExpenseOD message = null;
-            try {
-                message = controller.expenseInMonth(nameInput.getText(), valueInput.getText(), selectedCategory,
-                        isRecurring, monthID, currSession, selectedExpense);
-                // Update Month Menu
+        if (evt.getSource() == categoryCombo) {
+            this.selectedCategory = (String) categoryCombo.getSelectedItem();
+        } else if (selectedCategory == null){
+            JOptionPane.showMessageDialog(this, "Please select the previous category if you " +
+                "don't wish to edit. Thanks.");}
```

Comment:
Same for this, it should go inside the `if (evt.getSource() == categoryCombo)` clause.

Reply:
resolve as not relevant